unstructured/partition/xml.py

```diff
@@ -71,13 +71,11 @@ def partition_xml(
 
     if xml_keep_tags:
         if filename:
-            with open(filename) as f:
-                raw_text = f.read()
+            _, raw_text = read_txt_file(filename=filename, encoding=encoding)
         elif file:
-            f = spooled_to_bytes_io_if_needed(  # type: ignore
-                cast(Union[BinaryIO, SpooledTemporaryFile], file),
-            )
-            raw_text = f.read().decode(encoding)  # type: ignore
+            _, raw_text = read_txt_file(file=file, encoding=encoding)
```

Comment:
is there harm in keeping `spooled_to_bytes_io_if_needed`? in practice, this was introduced to work around a `SpooledTemporaryFile`  that could have originated, say as an `UploadFile` in FastApi

Reply:
I updated the `read_txt_file` utility function to keep using `spooled_to_bytes_io_if_needed` for xml